vllm/attention/layer.py

```diff
@@ -155,6 +159,13 @@ def forward(
         key: torch.Tensor,
         value: torch.Tensor,
     ) -> torch.Tensor:
+        """
+        The KV cache is stored inside this class and is accessed via
+        `self.kv_cache`.
+
+        Attention metadata (`attn_metadata`) is accessed via forward context
+        using `vllm.forward_context.get_forward_context().attn_metadata`.
```

Comment:
mention when we set it? (before excuting model forward)

Reply:
Done